h/static/scripts/reducers/annotations.js

```diff
@@ -272,6 +272,17 @@ function annotationExists(state, id) {
   });
 }
 
+function findIDsForTags(state, tags) {
```

Comment:
Couple of things here:

- This needs a doc comment.
- `findByTag` and `findByID` both have the signature `(annotations, XXX)` rather than `(state, XXX)`, and this should probably be the same.
- This should probably live next to `findByTag` and `findByID`?

More generally, I'm still concerned that we keep adding O(N^2) lookup functions... this is just going to come and bite us in the future on pages with lots of annotations.